civis/resources/_resources.py

```diff
@@ -19,45 +19,19 @@
 
 _RESOURCES = frozenset({"base", "all"})
 API_VERSIONS = frozenset({"1.0", })
-BASE_RESOURCES_V1 = [
-    'aliases',
-    'announcements',
-    'apps',
-    'civis',
-    'clusters',
-    'codes',
-    'credentials',
-    'databases',
-    'endpoints',
-    'enhancements',
-    'exports',
-    'files',
-    'git_repos',
-    'groups',
-    'imports',
-    'jobs',
-    'json_values',
-    'match_targets',
-    'media',
-    'models',
-    'notebooks',
-    'notifications',
-    'ontology',
-    'predictions',
-    'projects',
-    'queries',
-    'remote_hosts',
-    'reports',
-    'results',
-    'scripts',
-    'search',
-    'services',
-    'storage_hosts',
-    'tables',
-    'templates',
-    'users',
-    'workflows'
-]
+
+# civis_api_spec.json can be updated
+# by running the tools/update_civis_api_spec_json.py script.
+API_SPEC_PATH = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)),
+    "civis_api_spec.json",
+)
+with open(API_SPEC_PATH) as f:
+    API_SPEC = json.load(f, object_pairs_hook=OrderedDict)
+BASE_RESOURCES_V1 = sorted(
+    set(path.split("/", 2)[1] for path in API_SPEC["paths"].keys())
+)
```

Comment:
Practically, `BASE_RESOURCES_V1` is now a reference list of the publicly available API endpoints, used only by the Sphinx docs to auto-generate the API resources documentation (see `docs/source/conf.py`). As I've just updated `civis_api_spec.json`, this list now looks like the following:

```
In [1]: from civis.resources._resources import BASE_RESOURCES_V1

In [2]: BASE_RESOURCES_V1
Out[2]: 
['admin',
 'aliases',
 'announcements',
 'clusters',
 'credentials',
 'databases',
 'endpoints',
 'enhancements',
 'exports',
 'feature_flags',
 'files',
 'git_repos',
 'groups',
 'imports',
 'jobs',
 'json_values',
 'match_targets',
 'media',
 'models',
 'notebooks',
 'notifications',
 'ontology',
 'permission_sets',
 'predictions',
 'projects',
 'queries',
 'remote_hosts',
 'reports',
 'roles',
 'saml_service_providers',
 'scripts',
 'search',
 'services',
 'storage_hosts',
 'table_tags',
 'tables',
 'templates',
 'users',
 'workflows']
```